lib/active_merchant/billing/gateways/stripe.rb

```diff
@@ -426,6 +445,10 @@ def json_error(raw_response)
       def non_fractional_currency?(currency)
         CURRENCIES_WITHOUT_FRACTIONS.include?(currency.to_s)
       end
+
+      def emv_payment?(payment)
```

Comment:
Maybe a method on `CreditCard` that can tell you if it's an EMV card, and then this method to do the checking?